fix: redirect to api /signin from client /settings

ojeytonwilliams opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

This does create slightly strange behaviour if you sign out while on /settings. Specifically, you're immediately sent to /signin on the api and signed back in. @ahmaxed and I have been discussing a few solutions for this and will create a separate PR for them.

Closes https://github.com/freeCodeCamp/freeCodeCamp/issues/61897

<!-- Feel free to add any additional description of changes below this line -->

ojeytonwilliams committed:
fix: redirect to api /signin from client /settings

raisedadead reviewed (changes requested):
I would hold of merging this and including the correct redirection that you and Ahmad are discussing. It's better to have a solid UX than the loop (which is even confusing behavior).

ojeytonwilliams committed:
fix: navigate to learn, then sign out

ojeytonwilliams committed:
refactor: navigate to useEffect

It is a side effect, after all

ojeytonwilliams committed:
fix: simplify tests + use testing-library

Also, to let us spy on hardGoTo, I used window.location (which is mockable), not location (which isn't)

ojeytonwilliams committed:
refactor: drop unused eslint-disables

It's not clear why these aren't required, but it seems happy
